android/V8Runtime_shared.cpp

```diff
@@ -812,6 +812,54 @@ namespace facebook { namespace v8runtime {
     return handle_scope.Escape(v8BigIntValue->v8BigInt_.Get(v8::Isolate::GetCurrent()));
   }
 
+  jsi::BigInt V8Runtime::createBigIntFromInt64(int64_t val) {
+    return make<jsi::BigInt>(makeBigIntValue(v8::BigInt::New(GetIsolate(), val)));
+  }
+
+  jsi::BigInt V8Runtime::createBigIntFromUint64(uint64_t val) {
+    return make<jsi::BigInt>(makeBigIntValue(v8::BigInt::NewFromUnsigned(GetIsolate(), val)));
+  }
+
+  bool V8Runtime::bigintIsInt64(const jsi::BigInt&) {
+    // V8 doesn't internally track it
+    return true;
+  }
+
+  bool V8Runtime::bigintIsUint64(const jsi::BigInt& val) {
+    bool lossless {true};
+    bigIntRef(val)->Uint64Value(&lossless);
+    return lossless;
+  }
+
+  uint64_t V8Runtime::truncate(const jsi::BigInt& val) {
+    return bigIntRef(val)->Uint64Value(nullptr);
+  }
+
+  jsi::String V8Runtime::bigintToString(const jsi::BigInt&, int) {
+    throw jsi::JSINativeException("V8Runtime::bigintToString is not implemented!");
+  }
+
+  bool V8Runtime::hasNativeState(const jsi::Object& obj) {
+    return objectRef(obj)->InternalFieldCount() == 1;
+  }
+
+  std::shared_ptr<jsi::NativeState> V8Runtime::getNativeState(const jsi::Object& obj) {
+    std::shared_ptr<jsi::NativeState>* holder = static_cast<std::shared_ptr<jsi::NativeState>*>(objectRef(obj)->GetAlignedPointerFromInternalField(0));
```

Comment:
>0 [](http://example.com/codeflow?start=147&length=1)

It would be neater to define a constant/MACRO ..

Reply:
I just copied the non-android impl. No idea what that zero represents.

Reply:
Essentially, every v8 objects provide a small number of data slots (0 to some n) for the host/embedder to store private data ..  

It is ok to leave as is ..